packages/react-instantsearch-core/src/connectors/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,220 @@
+
+See https://www.algolia.com/doc/api-reference/api-parameters/optionalFilters/`
+      );
+    }
+
+    return acc;
+  }, []);
+
+  return props.transformSearchParameters(
+    new algoliasearchHelper.SearchParameters({
+      // @ts-ignore @TODO algoliasearch-helper@3.0.1 will contain the type
+      // `sumOrFiltersScores`.
+      // See https://github.com/algolia/algoliasearch-helper-js/pull/753
+      sumOrFiltersScores: true,
+      filters: `NOT objectID:${props.hit.objectID}`,
+      optionalFilters,
+    })
+  );
+}
+
+interface ConnectorState {
+  _searchParameters: PlainSearchParameters;
+}
+
+export default createConnector({
+  displayName: 'AlgoliaConfigureRelatedItems',
+
+  defaultProps,
+
+  getProvidedProps() {
+    return {};
+  },
+
+  getSearchParameters(
+    searchParameters: SearchParameters,
+    props: ConnectedProps<InternalConfigureRelatedItemsProps>
+  ) {
+    return searchParameters.setQueryParameters(
+      getSearchParametersFromProps(props)
+    );
+  },
+
+  transitionState(
+    this: ConnectorState,
+    props,
+    _prevSearchState,
+    nextSearchState
+  ) {
+    const id = getId();
+    // We need to transform the exhaustive search parameters back to clean
+    // search parameters without the empty default keys so we don't pollute the
+    // `configure` search state.
+    const searchParameters = removeEmptyArraysFromObject(
+      removeEmptyKey(getSearchParametersFromProps(props))
+    );
+
+    const searchParametersKeys = Object.keys(searchParameters);
+    const nonPresentKeys = this._searchParameters
+      ? Object.keys(this._searchParameters).filter(
+          prop => searchParametersKeys.indexOf(prop) === -1
+        )
+      : [];
+    this._searchParameters = searchParameters;
+    const nextValue = {
+      [id]: {
+        ...omit(nextSearchState[id], nonPresentKeys),
+        ...searchParameters,
+      },
+    };
+
+    return refineValue(nextSearchState, nextValue, {
+      ais: props.contextValue,
+      multiIndexContext: props.indexContextValue,
+    });
+  },
+
+  cleanUp(this: ConnectorState, props, searchState) {
+    const id = getId();
+    const indexId = getIndexId({
+      ais: props.contextValue,
+      multiIndexContext: props.indexContextValue,
+    });
+
+    const subState =
+      hasMultipleIndices({
+        ais: props.contextValue,
+        multiIndexContext: props.indexContextValue,
+      }) && searchState.indices
+        ? searchState.indices[indexId]
+        : searchState;
+
+    const configureKeys =
+      subState && subState[id] ? Object.keys(subState[id]) : [];
+
+    const configureState = configureKeys.reduce((acc, item) => {
+      if (!this._searchParameters[item]) {
+        acc[item] = subState[id][item];
+      }
+
+      return acc;
+    }, {});
+
+    const nextValue = { [id]: configureState };
+
+    return refineValue(searchState, nextValue, {
+      ais: props.contextValue,
+      multiIndexContext: props.indexContextValue,
+    });
+  },
```

Comment:
The implementation is mostly taken from:

https://github.com/algolia/react-instantsearch/blob/621656a8f25377094c6a3b06debc8d43f4e3d6c6/packages/react-instantsearch-core/src/connectors/connectConfigure.js#L38-L68